core/src/avm2/e4x/is_xml_name.rs

```diff
@@ -0,0 +1,443 @@
+// NOTE: Regex to match character groups: '\[#(.{5})-#(.{5})\]', substitution 'buf.extend(0$1..=0$2);\n'
+//       and regex to match single characters: '#(.{5})', substitution 'buf.push(0$1);\n'
+
+#![allow(clippy::vec_init_then_push)] // NOTE: This lint is not very helpful here.
+
+use crate::avm2::{Activation, Error, Value};
+use std::sync::OnceLock;
+
+// https://www.w3.org/TR/2004/REC-xml-20040204/#NT-Letter
+fn generate_letters() -> Box<[char]> {
+    let mut buf = vec![];
+
+    // Base char
+    buf.extend(0x0041..=0x005A);
```

Comment:
Two main questions:
- can we avoid making a 30kB vec (and iterating it) just for this? avmplus kinda got around it by only storing start-end ranges instead.
- any chance these ranges overlap with unicode, so we could use an existing (std)lib function to check for an unicode category?

Reply:
For the record: https://github.com/adobe/avmplus/blob/858d034a3bd3a54d9b70909386435cf4aec81d21/core/AvmCore.cpp#L3503

Reply:
And with ranges alone, the arrays are small enough that we could probably have them be just `const` instead of lazy-initialized.

Reply:
> Two main questions:
> 
>     * can we avoid making a 30kB vec (and iterating it) just for this? avmplus kinda got around it by only storing start-end ranges instead.
> 
>     * any chance these ranges overlap with unicode, so we could use an existing (std)lib function to check for an unicode category?

To be honest I do not know if any overlap with unicode, haven't looked into it.

Reply:
Pushed a commit to change it to static arrays instead.

Reply:
>     * any chance these ranges overlap with unicode, so we could use an existing (std)lib function to check for an unicode category?

There are methods with which ranges overlap, but if we would use them it would broaden the allowed characters, so compatibility with Flash would be lost. The reason for this is that Flash's characters were based on Unicode database version 2, but Rust's methods are based on the latest version.